lib/manager/dataset_manager.js

```diff
@@ -436,6 +461,212 @@ class DatasetManager {
+                                        'arrivalPageTime': currentPage
+                                    }
+                                    await writeFile(tripPath + '/pages.json', JSON.stringify(currentTrip));
+                                }
+                                // Avoid to add double page URI entries
+                                else if(currentTrip['arrivalPageTime'] !== currentPage && currentTrip['departurePageTime'] !== currentPage) {
+                                    // Add new page URI to the list and write it to disk
+                                    let currentDepartureTime = new Date(currentTrip['departurePageTime']);
+                                    let currentArrivalTime = new Date(currentTrip['arrivalPageTime']);
+                                    let newPageTime = new Date(currentPage);
+                                    
+                                    currentTrip = {
+                                        'departurePageTime': currentTrip['departurePageTime'],
+                                        'arrivalPageTime': currentTrip['arrivalPageTime']
+                                    };
+
+                                    if(newPageTime.getTime() < currentDepartureTime.getTime()) {
+                                        currentTrip['departurePageTime'] = currentPage;
+                                    }
+                                    
+                                    if(newPageTime.getTime() > currentArrivalTime.getTime()) {
+                                        currentTrip['arrivalPageTime'] = currentPage;
+                                    }
+                                    
+                                    await writeFile(tripPath + '/pages.json', JSON.stringify(currentTrip));
+                                }
+                            }
+                        }
+
+                        // Finished looping through pages, resolve Promise and show benchmark
+                        if(i == sets.length - 1) {
+                            t1 = (new Date().getTime() - t0) / 1000;
+                            logger.info('Vehicle index updated successfully (took ' + t1 + ' seconds)');
+                            resolve();
+                        }
+                    });
+                }
+            });
+        });
+    }
+
+    /**
+     * Cancel a complete trip based on tripURI and starting connection.
+     * The starting connection is the first connection from where the vehicle is cancelled.
+     * If the vehicle is completely cancelled, you can omit this parameter..
+     *
+     * Example: `await this.cancelTrip('http://irail.be/vehicle/IC106/20190301', 'sncb');`
+     * @param tripURI
+     * @param companyName
+     * @param startingConnection = null
+     * @return `true` if successful
+     */
+    async cancelTrip(tripURI, companyName, startingConnection = null) {
```

Comment:
If I understand correctly, in this method you will loop over the static pages changing the connections to `CancelledConnections`s of a certain trip. I would try to follow the same approach used for the real-time updates, where these changes are stored in separated files, keeping the static files intact and facilitating the memento process.

Reply:
Yeah that's even better! I will look into the RT files and see how I can accomplish this.

Reply:
Should be fixed now!